File: backend/src/schema/message_schema.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class MessageCreateRequest(BaseModel):
    channel_id: UUID
    user_id: UUID
    type: str
    content: str
    referenced_message_id: UUID | None
    model_config = ConfigDict(from_attributes=True)


class MessageResponse(BaseModel):
    id: UUID
    channel_id: UUID
    user_id: UUID
    type: str
    content: str
    referenced_message_id: UUID | None
    created_at: datetime
    updated_at: datetime
    model_config = ConfigDict(from_attributes=True)

File: frontend/src/schemas/messageSchema.ts
export interface Message {
  id: string;
  guil_id: string;
  name: string;
  messages: {
    id: string;
    channel_id: string;
    user_id: string;
    type: string;
    content: string;
    reference_id?: string | null;
    created_at: string;
    updated_at: string;
  }[];
}

export type MessagesResponse = Message;
